src/azure_hl.rs

```diff
@@ -163,23 +163,53 @@ pub enum CompositionOp {
     Count,
 }
 
+#[repr(i32)]
+#[derive(Clone, PartialEq)]
+pub enum JoinStyle {
+    Bevel = 0,
+    Round = 1,
+    Miter = 2,
+    MiterOrBevel = 3,
+}
+
+impl JoinStyle {
+    fn as_azure_join_style(self) -> AzJoinStyle {
+        self as AzJoinStyle
+    }
+}
+
+#[repr(i32)]
+#[derive(Clone, PartialEq)]
+pub enum CapStyle {
+    Butt = 0,
+    Round = 1,
+    Square = 2,
+}
+
+impl CapStyle {
+    fn as_azure_cap_style(self) -> AzCapStyle {
+        self as AzCapStyle
+    }
+}
+
 #[allow(non_snake_case)]
 pub struct StrokeOptions {
     pub line_width: AzFloat,
     pub miter_limit: AzFloat,
-    pub mDashPattern: *mut AzFloat,
+    pub mDashPattern: *const AzFloat,
```

Comment:
I don't know this code, but could we store an &'a [AzFloat] instead? Then we'd have some protection against uafs.